File: Algorithms/Swift/Easy/283_Move_Zeroes.swift
class Solution {
    func moveZeroes(inout nums: [Int]) {
        var i = 0, j = 0
        while j < nums.count {
            if nums[j] != 0 {
                nums[i] = nums[j]
                i += 1
            }
            j += 1
        }

        while i < nums.count {
            nums[i] = 0
            i += 1
        }
    }
}

File: Algorithms/JavaScript/Easy/283_Move_Zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    if (nums.length === 0) {
        return
    }

    var i = 0
    for (var j = 0; j < nums.length; j++) {
        if (nums[j] !== 0) {
            nums[i] = nums[j]
            i++
        }
    }

    while (i < nums.length) {
        nums[i++] = 0
    }
};
